clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v2/op/manifest/KubernetesDeployManifestOperation.java

```diff
@@ -262,11 +262,13 @@ private void applyTraffic(KubernetesManifestTraffic traffic, KubernetesManifest
   }
 
   private void validateManifestsForRolloutStrategies(List<KubernetesManifest> manifests) {
-    if (description.getStrategy() != null
-        && (manifests.size() != 1
-            || !manifests.get(0).getKind().equals(KubernetesKind.REPLICA_SET))) {
+    List<KubernetesManifest> replicaSetManifests =
+        manifests.stream()
+            .filter(m -> m.getKind().equals(KubernetesKind.REPLICA_SET))
+            .collect(Collectors.toList());
```

Comment:
Not worth changing here (as this isn't performance-sensitive code) but this could have been `.count()` as the terminal operation to directly populate a `long numReplicaSets` if we don't actually need a list of the replicatsets.

Reply:
Oh that is much nicer, I changed it!